### PR TITLE
Add the Open Data facet

### PR DESCRIPTION
**Pull request by coverbeck**
**Description**
Surface the Open Data facet. 

The facets column is getting really long, I wonder if it should start out collapsed by default? But that's for another issue/discussion.

Screen Shot 2023-01-31 at 5 17 04 PM

Screen Shot 2023-01-31 at 5 12 09 PM

**Review Instructions**
Verify that the open data facets are present as in the screenshots, and that clicking a search result when the `Has public test data` checkbox is selected, takes you to a workflow with an open data test param file.

**Issue**
dockstore/dockstore#5136

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

### CODE REVIEW

**Comment by coverbeck** [comment]
Failing test is the flaky one, I think.

File: src/app/search/map-friendly-values.pipe.ts [hunk 1]
```diff
@@ -112,6 +112,13 @@ export class MapFriendlyValuesPipe implements PipeTransform {
         ['INSTALL', 'Install'],
       ]),
     ],
+    [
+      'openData',
+      new Map([
+        ['0', 'No public test data'],
+        ['1', 'Has public test data'],
```
Comment (kathy-t) [inline src/app/search/map-friendly-values.pipe.ts:119]: Minor, but should keep the casing the same as the other facets. `Verified` and `Has Checker Workflows` have lower case facets

**Review by kathy-t — APPROVED** [review]

**Review by KimberleyChong — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]

**Comment by KimberleyChong** [comment]
> The facets column is getting really long, I wonder if it should start out collapsed by default? But that's for another issue/discussion.

That's a good point, maybe we can do something like in the [mocks](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/5fc690e874e0c807afeb299b) where about half of them are closed and the most priority ones are kept open.

**Comment by coverbeck** [comment]
> > The facets column is getting really long, I wonder if it should start out collapsed by default? But that's for another issue/discussion.
> 
> That's a good point, maybe we can do something like in the [mocks](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/5fc690e874e0c807afeb299b) where about half of them are closed and the most priority ones are kept open.

Nice find on the mocks, created a followup ticket, dockstore/dockstore#5341

**Review by svonworl — APPROVED** [review]

**Pull request merged**